src/Bot.hs

```diff
@@ -203,12 +205,20 @@ builtinCommands =
     , ( "wiggle"
       , ( "Wiggle the tenticle (integration with https://github.com/tsoding/wiggle)"
         , transR (Identity . messageSender) $
-          cmapR (URI.encode . T.unpack . senderName) $
-          Reaction $ \(Identity name) -> do
-            request <- parseRequest [qms|http://localhost:8081/wiggle/{name}|]
-            void $ httpRequest request))
+          cmapR (URI.encode . T.unpack . senderName) wiggle))
     ]
 
+wiggle :: Reaction Identity String
+wiggle = Reaction $ \(Identity name) -> wiggle' name True
+  where
+    defaultTimeoutMillis = 10000
+    wiggle' name isRetry = do
+      request <- parseRequest [qms|http://localhost:8081/wiggle/{name}|]
+      response <- httpRequest request
+      let status = getResponseStatus response
+      when (status /= ok200) $
+        when isRetry $ timeout defaultTimeoutMillis $ wiggle' name False
```

Comment:
This doesn't seem to handle several pending wiggle queries from different users. I believe we need to organize some kind of queue.